src/bitcoin_utils.rs

```diff
@@ -61,7 +61,7 @@ fn sign_with_tweak(
     merkle_root: Option<TapNodeHash>,
 ) -> schnorr::Signature {
     use bitcoin::hashes::Hash;
-    SECP.sign_schnorr(
+    SECP.sign_schnorr_no_aux_rand(
```

Comment:
Please add a comment about why this is safe / when this is safe. An example is here:

```suggestion
    // SECURITY: Using deterministic signatures (no aux randomness) for reproducibility
    // in withdrawal verification. This is safe because:
    // 1. Each transaction creates a unique sighash (different UTXOs/amounts/destinations)
    // 2. Keys are used in controlled environments without side-channel exposure
    // 
    // Trade-off: Reduced protection against side-channel attacks (timing/power analysis)
    // Acceptable because: [ADD YOUR SPECIFIC REASON - e.g., "keys are single-use and 
    // signing happens in isolated environments"]
```